src/charm.py

```diff
@@ -184,73 +151,72 @@ def reconfigure(self, event):
 
         if (
             self.unit.is_leader()
-            and self.cluster.replica_set_initialized
-            and self.cluster.need_replica_set_reconfiguration()
+            and self.state.mongodb_initialized
+            and self.need_replica_set_reconfiguration()
         ):
-            self.mongo.reconfigure_replica_set(self.cluster.hosts)
-            self.on.replica_set_configured.emit(self.cluster.hosts)
+            self.mongo.reconfigure_replica_set(self.cluster_hosts)
         self.on_update_status(event)
         logger.debug("Running reconfigure finished")
 
     ##############################################
-    ########## CLUSTER EVENT HANDLERS ############
+    #               RELATIONS                    #
     ##############################################
 
-    def on_mongodb_started(self, event):
-        if not self.unit.is_leader() or self.standalone:
-            self.on_update_status(event)
-            return
-        logger.debug("Running on_mongodb_started")
-        if not self.cluster.replica_set_initialized:
-            self.unit.status = WaitingStatus("Initializing the replica set")
-            self.mongo.initialize_replica_set(self.cluster.hosts)
-            self.on.replica_set_configured.emit(self.cluster.hosts)
-
-        self.on.cluster_ready.emit()
-        self.on_update_status(event)
-        logger.debug("Running on_mongodb_started finished")
+    def on_database_relation_changed(self, event):
+        event.relation.data[self.unit]['replicated'] = str(self.is_joined)
+        event.relation.data[self.unit][
+            'replica_set_name'] = self.replica_set_name
+        event.relation.data[self.unit]['standalone_uri'] = "{}".format(
+            self.standalone_uri)
+        event.relation.data[self.unit]['replica_set_uri'] = "{}".format(
+            self.replica_set_uri)
 
     ##############################################
-    ############### PROPERTIES ###################
+    #               PROPERTIES                   #
     ##############################################
 
     @property
     def mongo(self):
         return Mongo(
-            standalone_uri=self.cluster.standalone_uri,
-            replica_set_uri=f"{self.cluster.replica_set_uri}?replicaSet={self.replica_set_name}",
+            standalone_uri=self.standalone_uri,
+            replica_set_uri=f"{self.replica_set_uri}?replicaSet={self.replica_set_name}",
         )
 
+    @property
+    def replica_set_uri(self):
+        uri = "mongodb://"
+        for i, host in enumerate(self.cluster_hosts):
+            if i:
+                uri += ","
+            uri += f"{host}:{self.port}"
+        uri += "/"
+        return uri
+
+    @property
+    def standalone_uri(self):
+        return f"mongodb://{self.model.app.name}:{self.port}/"
```

Comment:
This line is not Python 3.5 compatible. 
What about this:

```python
"mongodb://{self.model.app.name}:{self.port}/".format(self.model.app.name, self.port)
```

Reply:
Thanks. Will fix.